### ISSUE TITLE
Fog of uncertainty

### ISSUE THREAD

**Issue by alin256** [CodingTrain/Directional-Boring#21]
One thought I had would be something along the lines of a "fog of war" other games use, simulating real-time scanning, that only reveals a small area around or slightly ahead of the drill head, and you'd have to react appropriately to any suddenly-appearing obstacles if you can, given the limited turning radius of your drill head.  I assume there's scanning and mapping done  well ahead of time IRL, of course, but this might a fun twist.

_Originally posted by @aerger in https://github.com/CodingTrain/Directional-Boring/issues/4#issuecomment-1175367485_

**Comment by alin256** [comment]
I have implemented a naive fog of uncertainty which reveals some distance around the drill bit. (In reality, it would only be to the sides of the bit.) 

It looks cool, but it is completely unusable. So I hesitate with sending a pull request. 
@shiffman and @aerger, please share your thoughts: https://al-digital.no/Directional-Boring/

**Comment by alin256** [comment]
I managed to implement a naive [seismic reflection detection](https://en.wikipedia.org/wiki/Reflection_seismology) algorithm to estimate the positions of boulders.

With that, the game became quite different but fun again:
https://al-digital.no/Directional-Boring/

@shiffman should I create a pull request or do we consider this a diverged branch? 

![image](https://user-images.githubusercontent.com/7563037/177772643-7a9c69a0-787b-48a2-9e0c-fbb49be7f6bd.png)

**Comment by shiffman** [comment]
This is so cool! I'd be glad to accept this as a pull request, yes! You are also welcome to submit separately to the "showcase" on the challenge page too if you like! https://thecodingtrain.com/challenges/172-directional-boring

**Comment by alin256** [comment]
@shiffman, I added a DOI (on Zenodo) to this release in case anyone decides to cite it. (I am at the stage of my academic career where I still play the citation game)

Since I spent most of the working day figuring things out, I felt obligated to extend the license with the mansion to my employer.

I hope this is ok with you :)

I am considering making it into a journal publication down the line. Let me know if you would like to join or if you have any relevant insights on the matter.
